app/AvailablePlugin/EmailAddressWidget/Model/EmailAddressWidgetVerification.php

```diff
@@ -133,6 +138,54 @@ public function addEmailToPerson($token, $actorCoPersonId) {
       throw new RuntimeException($e->getMessage());
     }
   }
+  
+  /**
+   * Replace the email and add a history record
+   *
+   * @since  COmanage Registry v4.1.0
+   * @param  string $token     Token used for verification
+   * @return integer           ID of the EmailAddress Record
+   * @throws RuntimeException
+   */
+  
+  public function replaceEmailForPerson($token, $actorCoPersonId) {
+    // Retrieve the Verification record and the configuration
+    $rec = $this->getRecordToVerify($token);
+    
+    // Create the CO Person Email record for replacement
+    $emailAttrs = array(
+      'id' => $rec['EmailAddressWidgetVerification']['email_id'],
+      'mail' => $rec['EmailAddressWidgetVerification']['email'],
+      'type' => $rec['EmailAddressWidgetVerification']['type'],
+      'verified' => true,
+      'co_person_id' => $rec['EmailAddressWidgetVerification']['co_person_id']
+    );
+    
+    try {
+      $EmailAddress = ClassRegistry::init('EmailAddress');
+      if(!$EmailAddress->save($emailAttrs, array("provision" => true,
+                                                 "trustVerified" => true))) {
+        throw new RuntimeException(_txt('er.db.save'));
+      }
+      
+      $CoPerson = ClassRegistry::init('CoPerson');
+      // History Record for the new record + verification
+      $CoPerson->HistoryRecord->record($rec['EmailAddressWidgetVerification']['co_person_id'],
+        null,
+        null,
+        $actorCoPersonId,
+        ActionEnum::EmailAddressVerified,
+        _txt('pl.emailaddresswidget.rs.mail.updated.verified',
+          array($rec['EmailAddressWidgetVerification']['email'],
+                $rec['EmailAddressWidgetVerification']['email_id'])));
+      
+      $this->delete($rec['EmailAddressWidgetVerification']['id']);
+      return $EmailAddress->id;
+      // Delete the Verification Request table record and return
+    } catch(Exception $e) {
```

Comment:
probably you need to delete the verification on an Exception

Reply:
Code updated to delete the EmailAddressWidgetVerification table row on this exception.